src/block/Dirt.php

```diff
@@ -79,7 +80,7 @@ public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player
 
 			$item->pop();
 			$world->setBlock($down->position, VanillaBlocks::HANGING_ROOTS());
-			//TODO: bonemeal particles, growth sounds
+			$world->addParticle($this->position, new BoneMealParticle());
```

Comment:
You'll need to leave the todo for the sounds, since you didn't integrate those

Reply:
BoneMealParticle is both a particle and a sound. I didn’t hear any other sounds in vanilla and there is no information about this on the wiki.

Reply:
> I didn’t hear any other sounds in vanilla

Isn't the sound of using bonemeal being `item.bone_meal.use`?

Reply:
I'm not a fan of adding more dual particle/sound events that masquerade as particles. We already have problems with BlockBreakParticle because of this.

#5169 needs to be dealt with.